numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -528,28 +539,42 @@ PyArray_SearchsideConverter(PyObject *obj, void *addr)
 static int order_parser(char const *str, Py_ssize_t length, void *data)
 {
     NPY_ORDER *val = (NPY_ORDER *)data;
+    int is_exact = 0;
+
     if (length != 1) {
         return -1;
     }
+
     if (str[0] == 'C' || str[0] == 'c') {
         *val = NPY_CORDER;
-        return 0;
+        is_exact = str[0] == 'C';
     }
     else if (str[0] == 'F' || str[0] == 'f') {
         *val = NPY_FORTRANORDER;
-        return 0;
+        is_exact = str[0] == 'F';
     }
     else if (str[0] == 'A' || str[0] == 'a') {
         *val = NPY_ANYORDER;
-        return 0;
+        is_exact = str[0] == 'A';
     }
     else if (str[0] == 'K' || str[0] == 'k') {
         *val = NPY_KEEPORDER;
-        return 0;
-    }
-    else {
+        is_exact = str[0] == 'K';
+    } else {
```

Comment:
style: let's leave this line as it was,
```suggestion
    }
    else {
```